content/en/docs/Badges/MCH2022/software-development/esp_idf_getting_started.md

```diff
@@ -0,0 +1,125 @@
+---
+title: "ESP-IDF getting started"
+linkTitle: "ESP-IDF getting started"
+nodateline: true
+weight: 1
+---
+
+
+
+Getting started using esp-idf is slightly different between Linux, Windows and Mac:
+- [Go here if you use Linux,](#on-ubuntu-and-debian)
+- [Go here if you use Mac.](TEMPLATELINK)
+- [Go here if you use Windows,](TEMPLATELINK)
+
+
+### On Ubuntu and Debian Linux
+Open a terminal and install the folowing software:
+```bash
+sudo apt-get install git wget flex bison gperf python3 python3-venv cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0
+```
+
+### On Arch Linux
+Open a terminal and install the folowing software:
+```bash
+sudo pacman -S --needed gcc git make flex bison gperf python cmake ninja ccache dfu-util libusb
+```
+
+Now, download the template app:
+```bash
+git clone https://github.com/badgeteam/mch2022-template-app my_fancy_app_name
+cd my_fancy_app_name
+make install
```

Comment:
This crashes with:
```
Traceback (most recent call last):
  File "tools/webusb_push.py", line 2, in <module>
    from webusb import *
  File "/home/a2800276/projects/MCH2022/my_fancy_app_name/tools/webusb.py", line 2, in <module>
    import usb.core
ModuleNotFoundError: No module named 'usb'
make: *** [Makefile:22: install] Error 1
```

because it's missing `pyUSB`

We could add a `pip install pyUSB` into the instructions, but that does not seem particularly elegant, I'm not sure where this would be addressed ideally.  But leaving it as is will cause a lot of frustration. 

The compilation goes through fine, but there are a LOT of warnings, I'll try to address those issues in the relevant repos, it's not a documentation error.  My feeling is we should be particularly vigilant about warnings in order to be able to spot actual problems easier instead of being drowned in irrelevant stuff.